### PR TITLE
UI Improvements

### PR DESCRIPTION
**Pull request by nnazo**
I've made some global UI improvements to the site, and added an menu bar to the top of the site, and altered a few of the components to look nicer.

### CODE REVIEW

File: food-truck-frontend/src/components/layout/MenuDropdownComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      <IconButton color="inherit"
+                  aria-controls="bar-menu"
+                  aria-haspopup="true"
+                  onClick={openMenu}>
+        <MenuIcon/>
+      </IconButton>
+      <Menu id="bar-menu"
+            anchorEl={anchorEl}
+            keepMounted
+            open={Boolean(anchorEl)}
```
Comment (MicahSchiewe) [inline food-truck-frontend/src/components/layout/MenuDropdownComponent.tsx:30]: Can we not just specify `anchorEl != null` here? Not a big issue, just curious.

File: food-truck-frontend/src/components/layout/theme/ThemeSwitchComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React, {useState} from "react";
+import {IconButton} from "@material-ui/core";
+import {ThemeProps} from "../../theme/FoodTruckThemeProvider";
+import {Brightness4, BrightnessHigh} from "@material-ui/icons";
+
+function ThemeSwitchComponent(props: ThemeProps) {
+  const [isDark, setIsDark]: [boolean, any] = useState(props.data.isDark);
```
Comment (MicahSchiewe) [inline food-truck-frontend/src/components/layout/theme/ThemeSwitchComponent.tsx]: I'd personally say to completely delete the internal state of this component and give it a prop reflecting the store state through mapStateToProps (unless that fails completely for functional components) which it uses to determine the checked state of the component, as this is a minor degree of duplication. Not really that serious, though.

File: food-truck-frontend/src/components/theme/FoodTruckThemeProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    ...themeOptions,
+    palette: {
+      ...themeOptions.palette,
+      background: props.data.isDark ? darkBackground : lightBackground,
+      type: props.data.isDark ? 'dark' : 'light'
+    }
+  }));
+
+  useEffect(() => {
+    // Prevent infinitely loading the theme, causing the site to be unresponsive.
```
Comment (MicahSchiewe) [inline food-truck-frontend/src/components/theme/FoodTruckThemeProvider.tsx:136]: I'll confess, I'm a bit confused about this. Does loadTheme need to be called after the component's mounted? If not, why not just call it in-line here and thus drop the state, effect, etc.?

File: food-truck-frontend/src/components/theme/ThemeD2P.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  dispatch({
+    type: ThemeActionTypes.SWITCH_THEME,
+    payload: { isDark: isDark }
+  });
+}
+
+const mapDispatchToProps = (dispatch: Dispatch<ThemeAction>) => {
+  return {
+    switchTheme: () => {
+      return new Promise<void>(() => {
```
Comment (MicahSchiewe) [inline food-truck-frontend/src/components/theme/ThemeD2P.ts]: I don't think we need to use Promises here. Returning a promise via a dispatch like this is specifically for performing things like API calls that we really want to be asynchronous (if you've seen 'thunk' referenced elsewhere in the project, this is what it's for--plain vanilla Redux does not support this). However, if the action is quick and simple, and occurs on the local machine, there isn't really a good reason in my mind to use an asynchronous dispatch (a thunk).

File: food-truck-frontend/src/pages/_document.js [hunk 5]
```diff
@@ -1,21 +1,21 @@
 import React from 'react';
 import Document, {Head, Html, Main, NextScript} from 'next/document';
 import {ServerStyleSheets} from '@material-ui/core/styles';
-import {theme} from '../util/theme';
+// import {theme} from '../util/theme';
```
Comment (MicahSchiewe) [inline food-truck-frontend/src/pages/_document.js]: Was this supposed to be deleted, or left commented out.

File: food-truck-frontend/src/pages/_document.js [hunk 6]
```diff
@@ -1,21 +1,21 @@
...
 
 export default class MyDocument extends Document {
     render() {
         return (
             <Html lang="en">
                 <Head>
                     {/* PWA primary color */}
-                    <meta name="theme-color" content={theme.palette.primary.main}/>
+                    {/*<meta name="theme-color" content={theme.palette.primary.main}/>*/}
```
Comment (MicahSchiewe) [inline food-truck-frontend/src/pages/_document.js:11]: Same with this

File: food-truck-frontend/src/components/login/LoginComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+import {Grid, TextField, Typography} from '@material-ui/core'
+import {UserData} from "../../redux/user/UserReducer";
+import NextLink from 'next/link'
+import LinkButton from "../layout/LinkButton";
+import Router from "next/router";
+
+type State = {
+  resultText: string;
+  data: UserData;
+  loadUserFromBackend: () => Promise<void>;
```
Comment (MicahSchiewe) [inline food-truck-frontend/src/components/login/LoginComponent.tsx]: loadUserFromBackend in the state just is duplicating information you already have in the props. I see no reason to include it.

File: food-truck-frontend/src/components/login/LoginComponent.tsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+interface LoginProps {
+  data: UserData;
```
Comment (MicahSchiewe) [inline food-truck-frontend/src/components/login/LoginComponent.tsx:14]: I'm not seeing this prop being used anywhere, or its reflection in the state.

Also worth noting, I'm fairly sure I read once that having a state variable that's supposed to reflect a props variables is an antipattern (can't find the article though, sorry). Thus, if this is needed, it should probably be kept in one or the other, not both.

**Review by MicahSchiewe — COMMENTED** [review]
Overall looked pretty good, though I was confused on a couple of things.

File: food-truck-frontend/src/components/layout/MenuDropdownComponent.tsx (hunk 1 above)
Comment (nnazo) [reply food-truck-frontend/src/components/layout/MenuDropdownComponent.tsx:30]: This is the way they had it in the examples in the MaterialUI docs, so not sure.

Comment (MicahSchiewe) [reply food-truck-frontend/src/components/layout/MenuDropdownComponent.tsx:30]: If that's how they had it in the docs, then let's keep it that way. Just wasn't sure.

File: food-truck-frontend/src/components/layout/theme/ThemeSwitchComponent.tsx (hunk 2 above)
Comment (nnazo) [reply food-truck-frontend/src/components/layout/theme/ThemeSwitchComponent.tsx]: I tried doing that at one point and it became an issue back when I was using a `Switch`. Though now that I've switched to an `IconButton` I should be able to get rid of it.

File: food-truck-frontend/src/components/theme/FoodTruckThemeProvider.tsx (hunk 3 above)
Comment (nnazo) [reply food-truck-frontend/src/components/theme/FoodTruckThemeProvider.tsx:136]: It needs to be called in order to get the theme from `localStorage`, and `useEffect` prevents the undefined `localStorage` issue.

File: food-truck-frontend/src/components/theme/ThemeD2P.ts (hunk 4 above)
Comment (nnazo) [reply food-truck-frontend/src/components/theme/ThemeD2P.ts]: I thought there might have been some other reason you were doing things this way before, so I kept with it. Will change this.

File: food-truck-frontend/src/pages/_document.js (hunk 5 above)
Comment (nnazo) [reply food-truck-frontend/src/pages/_document.js]: Will delete.

File: food-truck-frontend/src/pages/_document.js (hunk 6 above)
Comment (nnazo) [reply food-truck-frontend/src/pages/_document.js:11]: Will delete.

File: food-truck-frontend/src/components/login/LoginComponent.tsx (hunk 7 above)
Comment (nnazo) [reply food-truck-frontend/src/components/login/LoginComponent.tsx]: Refactored this in the latest commit.

File: food-truck-frontend/src/components/login/LoginComponent.tsx (hunk 8 above)
Comment (nnazo) [reply food-truck-frontend/src/components/login/LoginComponent.tsx:14]: I think some of this code was already there when I touched the file. Either way, I need `loadUserFromBackend` so I think the `data` field will still get passed to the component by redux

File: food-truck-frontend/src/components/theme/ThemeD2P.ts (hunk 4 above)
Comment (MicahSchiewe) [reply food-truck-frontend/src/components/theme/ThemeD2P.ts]: I see. My bad for not being a bit clearer as to what I was doing.

File: food-truck-frontend/src/components/theme/ThemeD2P.ts [hunk 9]
```diff
@@ -0,0 +1,43 @@
+import {Dispatch} from "redux";
+import {ThemeAction, ThemeActionTypes} from "../../redux/theme/ThemeActions";
+
+function getStoredTheme(): boolean {
```
Comment (MicahSchiewe) [inline food-truck-frontend/src/components/theme/ThemeD2P.ts]: I'll confess, the name doesn't make much sense to me. The function looks like it's checking for dark mode, not getting the stored theme.

**Review by MicahSchiewe — APPROVED** [review]
Other than one function name not exactly making sense to me, everything looks great.

File: food-truck-frontend/src/components/theme/ThemeD2P.ts (hunk 9 above)
Comment (nnazo) [reply food-truck-frontend/src/components/theme/ThemeD2P.ts]: I'll change that really quick. It was meant to return a string initially but I changed it to a boolean.

Edit: changed it to `storedThemeIsDark`

**Review by MicahSchiewe — APPROVED** [review]
LGTM

**Review by AlejandroNavarro1 — COMMENTED** [review]
It looks good throughout. I like the dark mode.

**Pull request merged**